sql/src/main/java/io/crate/planner/TableStatsService.java

```diff
@@ -154,5 +155,57 @@ public long numDocs(TableIdent tableIdent) {
         }
         return -1;
     }
+
+    @VisibleForTesting
+    ThreadPool.Cancellable refreshScheduledTask() {
+        return refreshScheduledTask;
+    }
+
+    @VisibleForTesting
+    TimeValue initialRefreshInterval() {
+        return initialRefreshInterval;
+    }
+
+    @VisibleForTesting
+    TimeValue lastRefreshInterval() {
+        return lastRefreshInterval;
+    }
+
+    @VisibleForTesting
+    NodeSettingListener nodeSettingListener() {
+        return nodeSettingListener;
+    }
+
+    @VisibleForTesting
+    class NodeSettingListener implements NodeSettingsService.Listener {
```

Comment:
you could let the TableStatsService class implement the listener interface, so you could avoid the getter above just for testing